blockchain/state/iterator.go

```diff
@@ -185,6 +185,144 @@ func (it *NodeIterator) retrieve() bool {
 	return true
 }
 
+// CheckStateConsistencyParallel checks the consistency of all state/storage trie of given two state databases in parallel.
+func CheckStateConsistencyParallel(oldDB Database, newDB Database, root common.Hash, quitCh chan struct{}) error {
+	// Check if the tries can be called
+	_, err := oldDB.OpenTrie(root)
+	if err != nil {
+		return err
+	}
+	_, err = newDB.OpenTrie(root)
+	if err != nil {
+		return err
```

Comment:
How about leaving a log in order to identify which database returns the error?

Reply:
Reflected :)